docs/book/src/topics/multitenancy.md

```diff
@@ -175,6 +177,60 @@ spec:
     name: multi-tenancy-role
 ```
 
+### Examples
+
+This is a deployable example which uses the `AWSClusterRoleIdentity` "test-account-role" to assume into the `arn:aws:iam::123456789:role/CAPARole` role in the target account.
+This example assumes that the `CAPARole` has already been configured in the target account.
+
+Finally, we inform the `Cluster` to use our `AWSCluster`type to provision a cluster in the target account specified by the `identityRef` section.
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AWSClusterRoleIdentity
+metadata:
+  name: "test-account-role"
+spec:
+  allowedNamespaces: {} # matches all namespaces
+  roleARN: "arn:aws:iam::123456789:role/CAPARole"
+  sourceIdentityRef:
+    kind: AWSClusterControllerIdentity # use the singleton for root auth
+    name: default
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AWSCluster
+metadata:
+  name: "test-multi-tenant-workload"
+spec:
+  region: "eu-west-1"
+  identityRef:
+    kind: AWSClusterRoleIdentity
+    name: test-account-role
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: "test-multi-tenant-workload"
+spec:
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: AWSCluster
+    name: "test-multi-tenant-workload"
+```
```

Comment:
I might have skipped  a step but without defining the `AWSClusterControllerIdentity` resource I get:

```
E0314 22:09:48.715984       1 controller.go:317] controller/awscluster "msg"="Reconciler error" "error"="failed to create scope: failed to create aws session: Failed to get providers for cluster: AWSClusterControllerIdentity.infrastructure.cluster.x-k8s.io \"default\" not found\nsigs.k8s.io/cluster-api-provider-aws/pkg/cloud/scope.NewClusterScope\n\t/workspace/pkg/cloud/scope/cluster.go:73\nsigs.k8s.io/cluster-api-provider-aws/controllers.(*AWSClusterReconciler).Reconcile\n\t/workspace/controllers/awscluster_controller.go:131\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581" "name"="test-multi-tenant-workload" "namespace"="default" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="AWSCluster" 
```

Reply:
The other option seems to be the `AutoControllerIdentityCreator=true` feature gate https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/docs/book/src/topics/multitenancy.md#awsclustercontrolleridentity.

Reply:
Ah, good catch.  I'm going to guess that you don't have the feature flag of `AutoControllerIdentityCreator=true`, enabled on your cluster.

It gets called about in the [AWSCLusterControllerIdentity](https://cluster-api-aws.sigs.k8s.io/topics/multitenancy.html#awsclustercontrolleridentity) section.

I'll add some steps indicating that if you *do not* have the flag enabled, creating your own `AWSClusterControllerIdentity` is required.